scene/resources/packed_scene.cpp

```diff
@@ -627,6 +819,70 @@ Error SceneState::_parse_node(Node *p_owner, Node *p_node, int p_parent_idx, Has
 		nd.index = p_node->get_index();
 	}
 
+	// Assign local-id if necessary.
+	if (p_node == p_owner || p_node->get_owner() == p_owner) {
+		if (nd.local_id > 0) { // Local id exists, but it may have been inherited from an instanced scene,
+			// so check for duplicates and reset to assign new
+			for (const Node *owned : p_owner->data.owned) {
+				if (owned != p_node && owned->get_local_id() == nd.local_id) {
+					WARN_PRINT(vformat("Node '%s' has duplicate scene local id %d with Node '%s'. Resetting", p_node->get_name(), nd.local_id, owned->get_name()));
+					p_node->set_local_id(0);
+					nd.local_id = 0;
+					break;
+				}
+			}
+		}
+		if (nd.local_id == 0) { // Local id hasn't been assigned already (or was reset).
+			print_verbose(vformat("Generating new scene local id for Node '%s'.", p_node->get_name()));
+			int32_t allocated_id = get_scene_local_id_offset() + 1;
+			while (true) { // Find unassigned value.
+				if (allocated_id >= _I32_MAX) { // This case shouldn't happen, but to be safe.
```

Comment:
```suggestion
				if (allocated_id >= INT32_MAX) { // This case shouldn't happen, but to be safe.
```
Did you mean this constant?

Reply:
an oversight, yes, thx